### DISCUSSION TITLE
Dual thermostat for servo mixing valve

### DISCUSSION THREAD

**Discussion by pozsothereal** [ScratMan/HASmartThermostat#110] [Ideas]
Hi, first thx for your good work, i would use it with a servo like mixing valve to control the primary water temp of my floor heating. It needs to be a dual climate with both heating and cooling switch unit, for turning the servo cw and ccw. Now i done this by 2 climate with altered temp of -0.1°C on one of them, and using a helper automation to keep one switch off for sure when heating/cooling is in on state. Thx in advance! 
![image](https://user-images.githubusercontent.com/61325530/184630126-6f19a2f3-30fc-4168-a378-55c98a4e5276.png)

- id: floor_water_temp
  alias: floor_water_temp
  trigger:
  - platform: state
    entity_id: 'input_text.floor_water_temp'
  action:
  - service: climate.set_temperature
    data_template:
      entity_id: climate.floor_heating_water_thermostat_decreasing
      temperature: "{{ states('input_text.floor_water_temp') }}"
  - service: climate.set_temperature
    data_template:
      entity_id: climate.floor_heating_water_thermostat_increasing
      temperature: "{{ states('input_text.floor_water_temp') | float - 0.1 }}"
      
- id: floor_relay_control_1
  alias: relay controller 1
  trigger:
  - platform: state
    entity_id: 'input_boolean.floor_water_decreasing'
    from: 'off'
    to: 'on'
  action:
  - service: input_boolean.turn_off
    data:
      entity_id: 'input_boolean.floor_water_increasing'
      
- id: floor_relay_control_2
  alias: relay controller 2
  trigger:
  - platform: state
    entity_id: 'input_boolean.floor_water_increasing'
    from: 'off'
    to: 'on'
  action:
  - service: input_boolean.turn_off
    data:
      entity_id: 'input_boolean.floor_water_decreasing'

**Comment by mocklarn** [comment]
Hi and thank you! I was looking for a way to control a motor managing my shunt-valve for my heating system and stumbled upon this. I am already using the integration to control some floorheating-actuators successfully.
Could you please elaborate how you ensure that both of the outputs are never active simultaneously? You are talking about a helper automation, how do you implement it?

**Comment by pozsothereal** [comment]
Hi, the section started from 'id: floor_relay_control_1' is the exclusion automation to help keep only one climate output active at the same time.

**Comment by mocklarn** [comment]
Sorry, when i am looking at it now it is clear. I was mistaken it for the connection between the thermostat and the output, but i guess that configuration is placed somewhere else in your configuration.yaml. 

I was also thinking about the PWM-periods. Are all thermostat PWM periods starting at the same time (as long as the period length is the same)? I guess otherwise it can be problem. Assuming that the the temperature is correct, the period of the thermostats are 10 seconds and the servo is first closing 5 sec and then opening 5 sec instead of just standing still for 10 seconds?

Regarding your exclusion, simple and elegant! Think I will use it in my coming control of my shunt-motor. Thank you!

**Comment by pozsothereal** [comment]
No problem, btw my conf looks like this, the waiting period is manageable with the right pwm, cycle time and k values i think, try to start with this;)

```
- platform: smart_thermostat  #https://github.com/ScratMan/HASmartThermostat
  name: floor_heating_water_thermostat_decreasing
  heater: input_boolean.floor_water_decreasing
  target_sensor: sensor.28_00000d5e304b_temperature
  min_temp: 20
  max_temp: 40
#   boost_pid_off: true
  min_cycle_duration:
    seconds: 1
  min_off_cycle_duration:
    seconds: 1
#   sensor_stall:
#     seconds: 60
  cold_tolerance: 0
  hot_tolerance: 0
  ac_mode: true
#   target_temp: 'input_number.floor_water'
  keep_alive:
    seconds: 10
  initial_hvac_mode: cool
#   outdoor_sensor: sensor.outdoor_temperature
  sampling_period: 0
    # milliseconds: 100
  away_temp: 20
  kp : 100
  ki : 40
  kd : 60
  pwm : 00:00:04

- platform: smart_thermostat  #https://github.com/ScratMan/HASmartThermostat
  name: floor_heating_water_thermostat_increasing
  heater: input_boolean.floor_water_increasing
  target_sensor: sensor.28_00000d5e304b_temperature
  min_temp: 20
  max_temp: 40
#   boost_pid_off: true
  min_cycle_duration:
    seconds: 1
  min_off_cycle_duration:
    seconds: 1
#   sensor_stall:
#     seconds: 60
  cold_tolerance: 0
  hot_tolerance: 0.2
  ac_mode: false
#   target_temp: 'input_number.floor_water'
  keep_alive:
    seconds: 10
  initial_hvac_mode: heat
#   outdoor_sensor: sensor.outdoor_temperature
  sampling_period: 0
    # milliseconds: 100
  away_temp: 20
  kp : 100
  ki : 40
  kd : 60
  pwm : 00:00:04
  ```

**Comment by mocklarn** [comment]
Thank you for the guidance @pozsothereal, it was very helpful! It took me some time to understand that it was helpers You were using since i haven´t used such before. I have not tested the configuration to manipulate the motor but the "click" and indicated states of my relay is telling me that it is working as intended.

**Comment by pozsothereal** [comment]
Hi,

Happy to help you!

Viktor

mocklarn ***@***.***> (időpont: 2022. okt. 1., Szo, 21:56)
ezt írta:

**Comment by ScratMan** [comment]
Hello, I don't understand what is your issue ? Should this be changed to a discussion instead ?

**Comment by pozsothereal** [comment]
Hi, yes sure we can, the point is to have both switch in one climate.

Like this:
heater: switch.study_heater
cooler: switch.study_cooler

Viktor
> Hello, I don't understand what is your issue ? Should this be changed to a discussion instead ?

**Reply by ScratMan** [reply]
Hi, maybe changes in [2023.7.0-beta2](https://github.com/ScratMan/HASmartThermostat/releases/tag/2023.7.0-beta2) is of interest for you...

**Comment by pozsothereal** [comment]
Hi!

Thx for the suggestion and for your work, will take a look!

p.

Adrien Bonnat ***@***.***> (időpont: 2023. júl. 26., Sze,
23:22) ezt írta:
